subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/DefaultExecutionStateChangeDetector.java

```diff
@@ -61,10 +62,11 @@ public ExecutionStateChanges detectChanges(AfterPreviousExecutionState lastExecu
             thisExecution.getInputFileProperties(),
             "Input file",
             executable);
-        InputFileChanges directInputFileChanges = new DefaultInputFileChanges(
+        DefaultInputFileChanges allInputFileChanges = new DefaultInputFileChanges(
             lastExecution.getInputFileProperties(),
             thisExecution.getInputFileProperties());
-        InputFileChanges inputFileChanges = errorHandling(executable, caching(directInputFileChanges));
+        InputFileChanges nonIncrementalInputFileChanges = allInputFileChanges.nonIncrementalChanges(incrementalPropertyNames);
+        InputFileChanges incrementalInputFileChanges = errorHandling(executable, caching(allInputFileChanges.incrementalChanges(incrementalPropertyNames)));
```

Comment:
Why no error handling for the non-incremental changes?

Reply:
The non-incremental changing will be added to an `errorHandling`, `summarizing` changes later down. That's why I don't need the error handling here.